include/sys/txg.h

```diff
@@ -84,14 +85,27 @@ extern void txg_kick(struct dsl_pool *dp);
  * Wait until the given transaction group has finished syncing.
  * Try to make this happen as soon as possible (eg. kick off any
  * necessary syncs immediately).  If txg==0, wait for the currently open
- * txg to finish syncing.
+ * txg to finish syncing.  This may be interrupted due to an exiting pool.
+ *
+ * If desired, flags can be specified using txg_wait_synced_tx(), in case
+ * the caller wants to be interruptible.
  */
-extern void txg_wait_synced(struct dsl_pool *dp, uint64_t txg);
+typedef enum {
+	/* Reject the call with EINTR upon receiving a signal. */
+	TXG_WAIT_F_SIGNAL	= (1U << 0),
+	/* Reject the call with EAGAIN upon suspension. */
```

Comment:
What's the difference between "suspension", "an exiting pool", and "forced export"?